File: api-java/src/main/java/fr/huiitre/tools/modules/riot/valorant/api/ValorantBundleController.java
package fr.huiitre.tools.modules.riot.valorant.api;

import java.util.List;
import java.util.UUID;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import fr.huiitre.tools.modules.core.common.api.RequiredRole;
import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.riot.valorant.application.catalog.usecase.GetValorantBundleByAssetIdUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.catalog.usecase.GetValorantBundleUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.catalog.usecase.ListValorantBundlesUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.catalog.view.ValorantBundleView;

@RestController
@RequestMapping("/riot/valorant")
public class ValorantBundleController {

    private final ListValorantBundlesUseCase listBundlesUseCase;
    private final GetValorantBundleUseCase getBundleUseCase;
    private final GetValorantBundleByAssetIdUseCase getBundleByAssetIdUseCase;

    public ValorantBundleController(
            ListValorantBundlesUseCase listBundlesUseCase,
            GetValorantBundleUseCase getBundleUseCase,
            GetValorantBundleByAssetIdUseCase getBundleByAssetIdUseCase) {
        this.listBundlesUseCase = listBundlesUseCase;
        this.getBundleUseCase = getBundleUseCase;
        this.getBundleByAssetIdUseCase = getBundleByAssetIdUseCase;
    }

    @RequiredRole(RoleCode.READ_ONLY)
    @GetMapping("/bundles")
    public List<ValorantBundleView> listBundles() {
        return listBundlesUseCase.execute();
    }

    @RequiredRole(RoleCode.READ_ONLY)
    @GetMapping("/bundles/{id}")
    public ValorantBundleView getBundle(@PathVariable Long id) {
        return getBundleUseCase.execute(id);
    }

    @RequiredRole(RoleCode.READ_ONLY)
    @GetMapping("/bundles/by-asset/{assetId}")
    public ValorantBundleView getBundleByAssetId(@PathVariable UUID assetId) {
        return getBundleByAssetIdUseCase.execute(assetId);
    }
}

File: api/Modules/Riot/Valorant/Api/ValorantBundleController.cs
using Microsoft.AspNetCore.Mvc;
using Tools.Api.Modules.Riot.Valorant.Application.Catalog.Usecases;
using Tools.Api.Modules.Riot.Valorant.Application.Catalog.Views;

namespace Tools.Api.Modules.Riot.Valorant.Api;

// Catalogue des packs.
[ApiController]
[Route("riot/valorant")]
public class ValorantBundleController : ControllerBase
{
    [HttpGet("bundles")]
    public Task<List<ValorantBundleView>> ListBundles(
        [FromServices] ListValorantBundlesUseCase listValorantBundlesUseCase
    )
    {
        return listValorantBundlesUseCase.Execute();
    }

    [HttpGet("bundles/{id:long}")]
    public Task<ValorantBundleView> GetBundle(
        [FromServices] GetValorantBundleUseCase getValorantBundleUseCase,
        [FromRoute] long id
    )
    {
        return getValorantBundleUseCase.Execute(id);
    }

    [HttpGet("bundles/by-asset/{assetId:guid}")]
    public Task<ValorantBundleView> GetBundleByAssetId(
        [FromServices] GetValorantBundleByAssetIdUseCase getValorantBundleByAssetIdUseCase,
        [FromRoute] Guid assetId
    )
    {
        return getValorantBundleByAssetIdUseCase.Execute(assetId);
    }
}
